src/radiation/geodesics.hpp

```diff
@@ -35,17 +35,48 @@ void GetXSource(Real &Kcon0, Real &Kcon1, Real &Kcon2, Real &Kcon3, Real src[NDF
   src[3] = Kcon3 / Kcon0;
 }
 
-KOKKOS_INLINE_FUNCTION
-void GetKSource(Real &X0, Real &X1, Real &X2, Real &X3, Real &Kcov0, Real &Kcov1,
-                Real &Kcov2, Real &Kcov3, Real &Kcon0,
-                const Geometry::CoordSysMeshBlock &geom, Real source[4]) {
-  SPACETIMELOOP(mu) { source[mu] = 0.; }
+template <class Geom>
+KOKKOS_INLINE_FUNCTION void GetKSource(Real &X0, Real &X1, Real &X2, Real &X3,
+                                       Real &Kcov0, Real &Kcov1, Real &Kcov2, Real &Kcov3,
+                                       Real &Kcon0, Geom &geom, Real source[4]) {
+  if constexpr (std::is_same<PHOEBUS_GEOMETRY, Geometry::FMKS>::value) {
```

Comment:
Is this right? Why is it limited to FMKS?

Reply:
Good point, I think you are correct. The existing code doesn't work for the torus (clearly), and I had some notion of exposing compile-time spacetime symmetries here, but likely did it wrong.

Reply:
Oh I see... let's do the generic thing for now. But we can probably add some introspection for geometries regarding symmetry at a later date. So we don't have to have explicit things here.

Actually... why do explicit finite differences here rather than just call the christoffel symbol infrastructure?

Reply:
Minkowski should be straight K = 0 though? So maybe (pseudocode) `if constexpr MINKOWSKI { /* 0*/ } else { /* be explicit */}`?

And because I just ported from `nubhlight` :) How might I clean it up?

Reply:
> Minkowski should be straight K = 0 though? So maybe (pseudocode) if constexpr MINKOWSKI { /* 0*/ } else { /* be explicit */}?

Oh yes good point. :)

> And because I just ported from nubhlight :) How might I clean it up?

Fine for now. Let's make an issue though. Correct thing to do is to either provide relevant functions from the metric if they don't exist or call the christoffel symbol functions used for the geometric source terms.

Reply:
Code corrected, cleaned, and issue created #227